lib/spack/spack/cmd/install.py

```diff
@@ -31,10 +32,64 @@
 level = "short"
 
 
+def parse_use_buildcache(args):
+    bc_parsed = {}
+    bc_keys = ["package", "dependencies", ""]
+    bc_values = ["only", "never", "auto"]
+    kv_list = re.findall("([a-z]+)?:?([a-z]+)", args.use_buildcache)
+
+    # Verify keys and values
+    bc_map = {k: v for k, v in kv_list if k in bc_keys and v in bc_values}
+    if not len(kv_list) == len(bc_map):
+        unrecognized_args = [(k, v) for k, v in kv_list if k not in bc_keys or v not in bc_values]
+        tty.error("Unrecognized arguments passed to use-buildcache: %s" % unrecognized_args)
+        tty.error(
+            "Expected: --use-buildcache "
+            ":{[auto|only|never],[[package:[auto|only|never]],[dependencies:[auto|only|never]]]"
+        )
+        exit(1)
+        return None
+
+    package_use_buildcache = "auto"
+    dependencies_use_buildcache = "auto"
+    if "" in bc_map:
+        package_use_buildcache = bc_map[""]
+        dependencies_use_buildcache = bc_map[""]
+    if "package" in bc_map:
```

Comment:
This will permit passing `--use-buildcache auto,package:never` and use the auto for dependencies. It is of a shorthand because I for one never type dependencies correctly and it is a long keyword.

Reply:
Is the idea here that you could also say `--use-buildcache auto,dependencies:never`?  I think it's weird if it only works the other way.

Reply:
Both ways work, also not listing a "default" and only specifying one works.

so...
`--use-buildcache package=never` is the same as `--use-buildcache auto,package=never` or `--use-buildcache dependencies=auto,package=never`

or you could do
`--use-buildcache dependencies=only` which is the same as `--use-buildcache auto,dependencies=only` or `--use-buildcache package=auto,dependencies=only`

and so on. The short is, by default the option is "auto", or it is the specified default (ie. specified in `--use-buildcache` without a prefix), or it is overridden by a prefixed option, ie. `package:only` or `dependencies:never` etc.

I left it open to allow some flexibility on how to specify the option based on the way the user is reasoning on what they want. Not sure if that is correct or not, trying to toe the line between overly restrictive and annoying to use.